pyiceberg/table/__init__.py

```diff
@@ -3537,6 +3537,58 @@ def update_partitions_map(
             schema=table_schema,
         )
 
+    def files(self) -> "pa.Table":
+        import pyarrow as pa
+
+        files_schema = pa.schema([
+            pa.field('content', pa.int8(), nullable=False),
+            pa.field('file_path', pa.string(), nullable=False),
+            pa.field('file_format', pa.string(), nullable=False),
```

Comment:
What do you think about making this a categorical type, similar to what we did for the refs table #602

Reply:
@geruh I saw your earlier discussion with Fokko on this, agree with making file_format categorical!

Reply:
@Fokko I think we could also make ```last_updated_snapshot_id``` also categorical?